pkg/controller/devicetainteviction/device_taint_eviction.go

```diff
@@ -165,11 +165,7 @@ func (p pool) getTaintedDevices() []taintedDevice {
 			continue
 		}
 		for _, device := range slice.Spec.Devices {
-			if device.Basic == nil {
-				// Unknown device type, not supported.
-				continue
-			}
-			for _, taint := range device.Basic.Taints {
+			for _, taint := range device.Taints {
```

Comment:
The manual changes required for v1beta1 to v1beta2/v1 are:
- removal of `Basic` indirection in ResourceSlice
- adding `Exactly` indirection in requests

With hindsight, this is something that could have been done in 1.33 - but it's still fairly mechanical and failed in obvious ways if not done (compile error).